### PR TITLE
python3Packages.django_2_2: init at 2.2

### PR DESCRIPTION
**Pull request by lsix**
###### Motivation for this change

This introduces Django-2.2, the new LTR version of django. As it only
supports python3, python2 will continue to use django_1_11 as LTR
version as it is still supported until 2020.

We can decide to continue with 1.11 as main version of django for some time, but I prefer to pick the last stable one.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/django/2_2.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      geos = geos;
+      gdal = gdal;
+      extension = stdenv.hostPlatform.extensions.sharedLibrary;
+    })
+  ];
+
+  # patch only $out/bin to avoid problems with starter templates (see #3134)
+  postFixup = ''
+    wrapPythonProgramsIn $out/bin "$out $pythonPath"
+  '';
```
Comment (dotlambda) [inline pkgs/development/python-modules/django/2_2.nix]: I don't think this does what the comment says it does. Instead, it wraps a second time creating e.g. `..django-admin-wrapped-wrapped`.

File: pkgs/development/python-modules/django/2_2.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  version = "2.2";
+
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "06d1x1a4vqzwmv73sixpcf993fj2rxzrl61617hi8387zgj46dbw";
+  };
+
+  patches = stdenv.lib.optionals withGdal [
```
Comment (dotlambda) [inline pkgs/development/python-modules/django/2_2.nix]:
```suggestion
  patches = stdenv.lib.optional withGdal
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -2347,7 +2347,7 @@ in {
 
   crayons = callPackage ../development/python-modules/crayons{ };
 
-  django = self.django_1_11;
+  django = if isPy3k then self.django_2_2 else self.django_1_11 ;
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]:
```suggestion
  django = if isPy3k then self.django_2_2 else self.django_1_11;
```

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -2357,6 +2357,10 @@ in {
     gdal = self.gdal;
   };
 
+  django_2_2 = callPackage ../development/python-modules/django/2_2.nix {
+    gdal = self.gdal;
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Should not be necessary unless we explicitly want `pkgs.gdal`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by lsix** [comment]
@dotlambda I have addressed your review. Sorry for such a long delay.

For the time being, I keep django-1.11 as main version, so we can introduce 2.2 and later migrate the default version of django used.

**Pull request merged**